src/analytics/lambdas/load-data-workflow/load-manifest-to-redshift.ts

```diff
@@ -79,14 +78,14 @@ const redshiftDataApiClient = getRedshiftClient(REDSHIFT_DATA_API_ROLE_ARN);
 export const handler = async (event: LoadManifestEvent, context: Context) => {
   logger.debug('requestJson:', JSON.stringify(event, undefined, 2));
   logger.debug(`context.awsRequestId:${context.awsRequestId}`);
-  var appId = event.detail.appId;
+  let appId = event.detail.appId;
   const manifestFileName = event.detail.manifestFileName;
   const jobList = event.detail.jobList;
   logger.info(`appId:${appId}, manifestFileName:${manifestFileName}, jobList:${JSON.stringify(jobList)}`);
   /**
    * The appId will be used as the schema of Redshift, '.' and '-' are not supported.
    */
-  for (var i=0; i < appId.length; i++) {
+  for (let i=0; i < appId.length; i++) {
```

Comment:
What's the type of `appId`, string or array? It looks like a bug.

Reply:
```
/**
   * The appId will be used as the schema of Redshift, '.' and '-' are not supported.
   */
  for (let i=0; i < appId.length; i++) {
    appId = appId.replace('.', '_').replace('-', '_');
  }
```
The `replace()` method can only replace one character per call.

Reply:
Once you use regex pattern, it should replace all.

```ts
function replaceDotsWithUnderscores(string: string): string {
  const pattern = /\./g;
  const replacement = "_";
  return string.replace(pattern, replacement);
}
```

Reply:
done